pkgs/development/compilers/hip/default.nix

```diff
@@ -116,105 +65,124 @@ stdenv.mkDerivation (finalAttrs: {
     hash = "sha256-VL0vZVv099pZPX0J2pXPFvrhkVO/b6X+ZZDaD9B1hYI=";
   };
 
-  nativeBuildInputs = [ cmake python3 makeWrapper perl ];
-  buildInputs = [ libxml2 numactl libglvnd libX11 python3Packages.cppheaderparser ];
+  nativeBuildInputs = [
+    makeWrapper
+    cmake
+    perl
+    python3Packages.python
+    python3Packages.cppheaderparser
+  ] ++ lib.optionals buildDocs [
+    doxygen
+    graphviz
+    fontconfig
+  ];
+
+  buildInputs = [
+    numactl
+    libxml2
+    libX11
+    libglvnd
+  ];
+
   propagatedBuildInputs = [
-    clang
-    llvm
+    rocminfo
+  ] ++ lib.optionals useAMD [
+    rocm-thunk
     rocm-comgr
     rocm-device-libs
     rocm-runtime
-    rocm-thunk
-    rocminfo
+    rocm-opencl-runtime
+  ] ++ lib.optionals (!useAMD) [
+    cudatoolkit
   ];
 
+  # https://github.com/ROCm-Developer-Tools/hipamd/commit/be33ec55acc104a59d01df5912261d007c7f3ee9
   patches = [
-    (substituteAll {
-      src = ./hipamd-config-paths.patch;
-      inherit clang llvm hip;
-      rocm_runtime = rocm-runtime;
+    (fetchpatch {
+      url = "https://github.com/ROCm-Developer-Tools/hipamd/commit/be33ec55acc104a59d01df5912261d007c7f3ee9.patch";
+      hash = "sha256-eTC4mUIN1FwRce1n38uDOlITFL/vpcOhvnaZTo5R7lo=";
     })
   ];
 
-  prePatch = ''
-    sed -e 's,#!/bin/bash,#!${stdenv.shell},' \
-        -i src/hip_embed_pch.sh
-  '';
-
-  preConfigure = ''
-    export HIP_CLANG_PATH=${clang}/bin
-    export DEVICE_LIB_PATH=${rocm-device-libs}/lib
-  '';
-
   cmakeFlags = [
-    "-DHIP_PLATFORM=amd"
-    "-DAMD_OPENCL_PATH=${rocm-opencl-runtime.src}"
-    "-DHIP_COMMON_DIR=${hip}"
-    "-DROCCLR_PATH=${rocclr}"
-    "-DHIP_VERSION_BUILD_ID=0"
+    "-DROCM_PATH=${rocminfo}"
+    "-DHIP_PLATFORM=${hipPlatform}"
+    "-DHIP_OFFICIAL_BUILD=ON"
+    "-DHIP_COMMON_DIR=/build/source/hip-common"
+    "-DHIPCC_BIN_DIR=/build/source/hipcc/build"
```

Comment:
Please do not hardcode the build directory.

Reply:
Is there an alternative?

Reply:
It's ``src.name``, I saw it in one of the other derivations earlier.

Reply:
`src.name` has nothing to do with the absolute directory that will be dropped into during a build -- it can be anything, including not in `/build` when outside of a sandbox. Prefer `$PWD`, or relative paths when possible.